cartoframes/core/cartodataframe.py

```diff
@@ -4,36 +4,169 @@
+
+            .. code::
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('nybb.shp')
+                cdf.to_carto(if_exists='replace')
+        """
+
         from ..io.carto import to_carto
         return to_carto(self, *args, **kwargs)
 
     def convert(self, index_column=None, geom_column=None, lnglat_columns=None,
                 drop_index=True, drop_geom=True, drop_lnglat=True):
-        # Magic function
+        """ For internal usage only.
+        Tries to decode the geometry automatically as a `shapely https://pypi.org/project/Shapely/_.`
+        object by looking for coordinates in columns.
+
+        Args:
+            index_column (str, optional):
+                Name of the index column. If it is `None`, it is generated automatically.
+            geom_column (str, optional):
+                Name of the geometry column to be used to generate the decoded geometry.
+                If it is None, it tries to find common geometry column names, but if there is
+                no geometry column it will leave it empty.
+            lnglat_columns ([str, str], optional):
+                Tuple with the longitude and latitude column names to be used to generate
+                the decoded geometry.
+            drop_index (bool, optional):
+                Defaults to True. Removes the index column.
+            drop_geom (bool, optional):
+                Defaults to True. Removes the geometry column.
+            drop_lnglat (bool, optional):
+                Defaults to True. Removes the lnglat column.
+
+        Returns:
+            The CartoDataFrame itself
+
+        Examples:
+
+            Decode the geometry automatically:
+
+            .. code::
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('filename.csv').convert()
+
+            Passing the geometry column explicitly:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('filename.csv')
+                cdf.convert(geom_column='my_geom_column')
+
+            Passing lnglat_columns explicitly:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('filename.csv')
+                cdf.convert(lnglat_columns=['longitude', 'latitude'])
+
+            Passing the index column explicitly:
+
+            .. code::
+
+                from cartoframes import CartoDataFrame
+
+                cdf = CartoDataFrame.from_file('filename.csv')
+                cdf.convert(index_column='my_index')
+
+        """
+
         generate_index(self, index_column, drop_index)
         generate_geometry(self, geom_column, lnglat_columns, drop_geom, drop_lnglat)
         return self
 
     def viz(self, *args, **kwargs):
+        """
+        Creates a :py:class:`Map <cartoframes.viz.Map>`. visualization
```

Comment:
I think there is an extra dot here. I would comment that you can use the Layer API here.

Reply:
I wouldn't add anything else about the Layer API yet until we consolidate this, it's more an extra for now

Reply:
Yes, you're right